core/components/molecules/modal/Modal.tsx

```diff
@@ -217,6 +217,12 @@ class Modal extends React.Component<ModalProps, ModalState> {
       ['Overlay-container--open']: open,
     });
 
+    const bodyClass = classNames({
+      ['Modal-body']: true,
+      ['Modal-body--withMargin']: !!footer && open,
+      ['Modal-body--withPadding']: !footer && open,
```

Comment:
No need to add check for `open`.

Reply:
Condition should be `(headerOptions || footerOptions || footer || header) ? !!footer : true` to support both old and new api of Modal.